[m1oojv] iP

m1oojv opened the pull request:
# Sam - Your Ultimate Task Manager 👑

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Sam is the ultimate solution to 
- free your mind from the burden of remembering tasks
- has user-friendly text-based interface
- simplifies task management 
- offers lightning-fast performance : ~~FAST~~ _SUPER_ FAST to use

`System.out.printf("Download now")`

## Getting Started with Sam

1. **Download Sam**: Get started by [downloading Sam](https://www.dukepro-taskmanager.com/download) from our official website.
2. **Installation**: Simply double-click the downloaded file to install Sam on your device.
3. **Add Your Tasks**: Start adding your tasks, and let Sam efficiently manage them for you 😉.

And it is **FREE**! 

## Features:
- [X] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you're a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
} ```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

m1oojv committed:
Level 0. Rename, Greet, Exit: Give your chatbot a new name, Implement an initial skeletal version

m1oojv committed:
Level 1. Echo: echos commands entered by the user, and exits when the user types the command bye

m1oojv committed:
Level 1: Remove unnecessary \n commands

m1oojv committed:
Level 2. Add, List
Add the ability to store whatever text entered by the user and display them back to the user when requested.

m1oojv committed:
Level 3. Mark as Done
Add the ability to mark tasks as done. Optionally, add the ability to change the status back to not done.

m1oojv committed:
Level 4. ToDos, Events, Deadlines
Add support for tracking three types of tasks:
ToDos: tasks without any date/time attached to it
Deadlines: tasks that need to be done before a specific date/time
Events: tasks that start at a specific date/time and ends at a specific date/time

m1oojv committed:
Test using the I/O redirection technique

m1oojv committed:
Level 5. Handle Errors
Teach the chatbot to deal with errors such as incorrect inputs entered by the user.

m1oojv committed:
Level 6. Delete
Add support for deleting tasks from the list.

m1oojv committed:
Use Enumerations

m1oojv committed:
Level 7. Save
Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when the chatbot starts up. You may hard-code the file name and relative path from the project root e.g., ./data/duke.txt
Your code must handle the case where the data file doesn't exist at the start.
Stretch goal: Handle the situation of the data file being corrupted (i.e., content not in the expected format).

m1oojv committed:
Merge pull request #1 from m1oojv/branch-Level-7

Level 7. Save

m1oojv committed:
Update to Level 8. Dates and Times
Teach the chatbot how to understand dates and times. For example, if the command is deadline return book /by 2/12/2019 1800, the chatbot should understand 2/12/2019 1800 as 2nd of December 2019, 6pm, instead of treating it as just a String.

m1oojv committed:
Merge pull request #2 from m1oojv/branch-Level-8

Update to Level 8. Dates and Times

m1oojv committed:
Update files to practice better OOP and organize into packages

m1oojv committed:
Merge pull request #3 from m1oojv/branch-a-moreoop

Update files to practice better OOP and organize into packages

m1oojv committed:
Fix test files

m1oojv committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-a-gradle

m1oojv committed:
Update packages

m1oojv committed:
Update build gradle

m1oojv committed:
Merge pull request #4 from m1oojv/branch-a-gradle

A-Gradle

m1oojv committed:
Add JUnit tests

m1oojv committed:
Add JavaDoc

m1oojv committed:
Add JavaDoc for Storage

m1oojv committed:
Update code to fulfill Standards/Conventions

m1oojv committed:
Update code to fulfill Standards/Conventions

m1oojv committed:
Implement Level 9. Find
Give users a way to find a task by searching for a keyword.

m1oojv committed:
Merge pull request #5 from m1oojv/branch-Level-9

Implement Level 9. Find

m1oojv committed:
Merge branch 'master' into branch-A-CodingStandard

m1oojv committed:
Merge pull request #6 from m1oojv/branch-A-CodingStandard

Merge Branch A -coding-standard

m1oojv committed:
Resolve runtest.sh bugs of failed tests

m1oojv committed:
Add Level 10. GUI
Add a GUI to the chatbot. Use the JavaFX technology to implement the GUI.

m1oojv committed:
Merge pull request #7 from m1oojv/branch-Level-10

Add Level 10. GUI

richiehx commented on src/main/java/Sam/commands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,47 @@
+ */
+public class AddDeadlineCommand extends Command {
+
+    public static final String COMMAND_WORD = "deadline";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a deadline to the task list. \n"
```
Nice constant names! Perhaps you could include some comments on what the constants are for?

richiehx commented on src/main/java/Sam/commands/Command.java:
I like how you abstracted the separate commands into their own classes! Very nice 👍

richiehx commented on src/main/java/Sam/commands/HelpCommand.java:
I like the addition of the help command! Very Nice 👍

richiehx commented on src/main/java/Sam/constants/Message.java:
Interesting way of doing things! Very nice 👍

richiehx commented on src/main/java/Sam/services/Storage.java:
```diff
@@ -0,0 +1,67 @@
+        this.path = path;
+    }
+
+    /**
+     * Ensure that the hard disk is present in the file directory.
+     * Creates the file if not found
+     */
```
You might want to add the @return line to enhance your javadocs!

richiehx commented on src/main/java/Sam/services/parser/DateTimeParser.java:
I like how you separated the parser into their components! Very nice 👍

richiehx commented on src/main/resources/images/sam.gif:
Nice images! 👍

richiehx commented on src/test/java/sam/DeadlineTest.java:
Nice tests! 👍

richiehx reviewed (commented):
LGTM! I like how you separated the components.

kohkaijie commented on src/main/java/Sam/commands/AddEventCommand.java:
```diff
@@ -0,0 +1,48 @@
+
+    public static final String COMMAND_WORD = "event";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds an event to the task list. \n"
```
Great use of an example message to guide user to input the message in the correct format

kohkaijie commented on src/main/java/Sam/constants/Message.java:
```diff
@@ -0,0 +1,25 @@
+/**
+ * Container for user visible messages.
+ */
+public class Message {
```
Great use of class abstracction to keep track of possible messages!

kohkaijie commented on src/main/java/Sam/services/parser/DateTimeParser.java:
```diff
@@ -0,0 +1,46 @@
+            DateTimeFormatter.ofPattern("yyyy-M-d")
+    );
+
+    private static List<DateTimeFormatter> dateTimeFormatters = Arrays.asList(
```
Perhaps you can include more formats for time inputs?

kohkaijie commented on src/main/java/Sam/services/parser/FileParser.java:
```diff
@@ -0,0 +1,66 @@
+        String taskDescription = parts[2].trim();
+        String additionalInfo = parts.length > 3 ? parts[3].trim() : null;
+
+        switch (taskType) {
```
Great use of switch-case statements to prevent heavily nested loops!

kohkaijie reviewed (commented):
Overall, great use of abstraction to seperate your code into various functionalities and the coding standard is overall met with use of Egyptian braces and correct indentation. Keep up the good work!

m1oojv committed:
Update A-Assertions
There is no assertions in the code

Assertions are used to define assumptions about the program state so that the runtime can verify them.
Assertions serve a valuable purpose by aiding programmers
in verifying the accuracy of our assumptions and
ensuring that no invariants are breached.

We should incorporate assertions into the code in areas
where it's crucial to validate these assumptions.

m1oojv committed:
Merge pull request #8 from m1oojv/branch-A-Assertions

Update A-Assertions

m1oojv committed:
Add Comments to Message.java

m1oojv committed:
Update javadocs in Storage.java

m1oojv committed:
Update to correct some code quality violations

m1oojv committed:
Merge pull request #9 from m1oojv/branch-A-CodeQuality

Branch a code quality

m1oojv committed:
Feat: Add C-Tagging
There is no way to tag tasks in the code

Tagging tasks is a useful organizational feature
in task management systems. Tags allow us to
categorize tasks based on different criteria,
such as priority, project, context, or location.

Let's add this new feature of tagging tasks in our task list
for a better user experience

m1oojv committed:
Merge pull request #10 from m1oojv/branch-BCD-Extension

Feat: Add C-Tagging

m1oojv committed:
Minor updates to comments and tests

m1oojv committed:
Dummy names

m1oojv committed:
Correct naming of files

m1oojv committed:
Merge pull request #12 from m1oojv/branch-fix-naming

Fix: Branch fix naming of files

m1oojv committed:
Feat: Add User Guide

m1oojv committed:
Fix: Fix User Guide demo picture

m1oojv committed:
Fix: Fix Gradle build file

m1oojv committed:
chore: Fix Gradle build file

m1oojv committed:
style: Correct styling of certain javadocs

m1oojv committed:
feat: Add responsive height of chat bot and wrap text boxes

m1oojv committed:
feat: Update duke

m1oojv committed:
feat: Update text box of user to align right

m1oojv committed:
Merge pull request #13 from m1oojv/branch-gui-changes

feat: Update text box of user to align right
